### PR TITLE
Updated Memory Document

### PR DESCRIPTION
**Pull request by terrylucas**

### CODE REVIEW

File: src/docs/development/tools/devtools/memory.md [hunk 1]
```diff
@@ -5,41 +5,68 @@ description: Learn how to use the DevTools memory view.
...
-* Monitoring of Dart total instances,
+* Monitor of Allocations&mdash;total instances,
   size and accumulator monitoring to detect memory leaks
+* Track when memory for a particular class is contructed
+  (new instance)&mdash;stack trace of your code
 * Analysis of a Snapshot to detect possible memory problems
   or Flutter gotchas
-* Snapshots of all live memory class instances and sentinels
-  (instances to be GC'd)
+* Snapshot of all live memory class instances and sentinels
```
Comment (kenzieschmoll) [inline src/docs/development/tools/devtools/memory.md:30]: r/Snapshot/Snapshot and inspection

end sentence after GC'd).

File: src/docs/development/tools/devtools/memory.md [hunk 2]
```diff
@@ -5,41 +5,68 @@ description: Learn how to use the DevTools memory view.
...
+user fired events (using the ```dart:developer``` package) are collected
+in the same timeline as the memory statistics. All of this collected
+statistics and events are displayed in charts (see Anatomoy of Memory).
+
+The next level of understanding memory involes you directly interacting
+with DevTools and your application to isolate a short period of time that
+you are interested in knowing how many objects were allocated, how many
+bytes were allocated, or tracking all the places in your code where a
+particular class is allocated. This information is available under the
+"Allocations" tab of Memory profiler.
```
Comment (kenzieschmoll) [inline src/docs/development/tools/devtools/memory.md:49]: I think this whole section can be replaced with:

When the memory tab is opened for the first time in DevTools, memory statistics from the VM are collected. The memory usage data collected consists of total heap size used, external heap size used, maximum heap capacity, and resident set size (RSS). The usage statistics are polled and displayed in the memory overview chart.

As you interact with your application, various events are also detected and displayed in the overview chart (e.g. GC's, Flutter events, user-fired events from `dart:developer`).

File: src/docs/development/tools/devtools/memory.md [hunk 3]
```diff
@@ -5,41 +5,68 @@ description: Learn how to use the DevTools memory view.
...
+
+At the top-level, when the memory tab is selected memory statistics from
+the VM are collected. These statistics are displayed in the two overview
+charts (Dart memory and Android-only) the collection of general memory
+usage e.g., total heap used, external heap, maximum heap capacity,
+Resident Set Size (RSS). As you interact with your application various
+events are detected e.g., memory GC (garabage collection),Flutter events,
+user fired events (using the ```dart:developer``` package) are collected
+in the same timeline as the memory statistics. All of this collected
+statistics and events are displayed in charts (see Anatomoy of Memory).
```
Comment (kenzieschmoll) [inline src/docs/development/tools/devtools/memory.md:42]: I think this whole section can be replaced with:

When the memory tab is opened for the first time in DevTools, memory statistics from the VM are collected. The memory usage data collected consists of total heap size used, external heap size used, maximum heap capacity, and resident set size (RSS). The usage statistics are polled and displayed in the memory overview chart.

As you interact with your application, various events are also detected and displayed in the overview chart (e.g. GC's, Flutter events, user-fired events from dart:developer).

File: src/docs/development/tools/devtools/memory.md [hunk 4]
```diff
@@ -5,41 +5,68 @@ description: Learn how to use the DevTools memory view.
...
 
-Allocated Dart objects created using a class constructor
-(for example, by using `new MyClass()` or `MyClass()`)
-live in a portion of memory called the heap.
+Allocated Dart objects created using a class constructor (for
+example, by using `new MyClass()` or `MyClass()`) live in a partion
+of memory called the heap that is managed by the Dart VM (virtual
+machine). 
 
 ## DevTools memory page
```
Comment (kenzieschmoll) [inline src/docs/development/tools/devtools/memory.md:13]: delete this header and move the following info under what is it

File: src/docs/development/tools/devtools/memory.md [hunk 5]
```diff
@@ -5,41 +5,68 @@ description: Learn how to use the DevTools memory view.
...
-assets for thumbnail-sized images in inefficient, memory usage can
-be improved by loading smaller assets or adjusting the cacheWidth/cacheHeight
-to decode an image to a smaller size reducing the memory usage of the
-ImageCache. The analysis catches issues like this.
-
-Memory profiling consists of six parts, each increasing
-in granularity:
+memory at a given moment. 
+
+Memory profiling consists of six parts, each increasing in granularity:
```
Comment (kenzieschmoll) [inline src/docs/development/tools/devtools/memory.md:18]: I think we should condense this and the bullets below. 
```
Memory profiling in DevTools consists of 3 main functions:
* Monitor the memory overview chart for usage statistics and events
* View memory snapshots and analyses to detect possible memory issues and inspect objects
* Monitor and track allocations for specific classes and objects

## Memory Overview Chart
<insert all the information about the overview chart, including the memory usage statistics, events, etc.>

## Snapshot and Analysis
< insert all info about snapshot, analysis, etc. including any functions of the controls that are associated with this data >

## Allocation Monitoring and Tracking
< insert all info about monitoring and tracking allocations >
```

File: src/docs/development/tools/devtools/memory.md (hunk 4 above)
Comment (terrylucas) [reply src/docs/development/tools/devtools/memory.md:13]: Please review I've updated with the latest UX and re-organized the doc to have a better flow.

File: src/docs/development/tools/devtools/memory.md (hunk 5 above)
Comment (terrylucas) [reply src/docs/development/tools/devtools/memory.md]: I've reorganized the doc all the information was added but was a bit scattered. Better flow.

File: src/docs/development/tools/devtools/memory.md [hunk 6]
```diff
@@ -5,39 +5,64 @@ description: Learn how to use the DevTools memory view.
 
 ## What is it?
 
-Allocated Dart objects created using a class constructor
-(for example, by using `new MyClass()` or `MyClass()`)
-live in a portion of memory called the heap.
+Allocated Dart objects created using a class constructor (for
+example, by using `new MyClass()` or `MyClass()`) live in a partion
+of memory called the heap that is managed by the Dart VM (virtual
```
[the remark this answers is not in the document]
Comment (terrylucas) [reply src/docs/development/tools/devtools/memory.md:10]: Done.

File: src/docs/development/tools/devtools/memory.md [hunk 7]
```diff
@@ -5,39 +5,64 @@ description: Learn how to use the DevTools memory view.
 
 ## What is it?
 
-Allocated Dart objects created using a class constructor
-(for example, by using `new MyClass()` or `MyClass()`)
-live in a portion of memory called the heap.
+Allocated Dart objects created using a class constructor (for
+example, by using `new MyClass()` or `MyClass()`) live in a partion
```
[the remark this answers is not in the document]
Comment (terrylucas) [reply src/docs/development/tools/devtools/memory.md:9]: Done.

File: src/docs/development/tools/devtools/memory.md [hunk 8]
```diff
@@ -5,39 +5,64 @@ description: Learn how to use the DevTools memory view.
...
-in granularity:
+memory at a given moment. 
+
+Memory profiling consists of six parts, each increasing in granularity:
 
 * Memory overview chart
 * Android-only memory chart (via Android's ADB tool)
 * Event timeline )e.g., garbage collection (GC) events, user memory tooling interactions)
-* Monitoring of Dart total instances, size and accumulator monitoring to detect memory leaks
+* Monitoring of Allocations total instances, size and accumulator monitoring to detect memory leaks
```
[the remark this answers is not in the document]
Comment (terrylucas) [reply src/docs/development/tools/devtools/memory.md]: Re-organized and removed.

File: src/docs/development/tools/devtools/memory.md [hunk 9]
```diff
@@ -5,39 +5,64 @@ description: Learn how to use the DevTools memory view.
...
+* Tracking exactly where in your code a class constructor is called (memory allocated)
```
[the remark this answers is not in the document]
Comment (terrylucas) [reply src/docs/development/tools/devtools/memory.md]: Re-organized and removed.

File: src/docs/development/tools/devtools/memory.md (hunk 1 above)
Comment (terrylucas) [reply src/docs/development/tools/devtools/memory.md]: Reorged and removed.

**Comment by terrylucas** [comment]
PTAL doc is up to date with latest UX and was re-organized to have better flow.

**Comment by sfshaza2** [comment]
I really apologize, @terrylucas, I keep getting pulled into other things, but I will do this TODAY.

**Review by sfshaza2 — APPROVED** [review]
LGTM. I'm going to merge this and accept @terrylucas's changes (where they conflict) and then take another look.

**Pull request merged**